react/components/admin/pages/Form/index.tsx

```diff
@@ -267,23 +279,30 @@ class FormContainer extends Component<Props, State> {
                 declarer,
                 domain,
                 interfaceId,
-                metaTags: {
-                  description: metaTagDescription,
-                  keywords: (metaTagKeywords || []).map(({ value }) => value),
-                },
-                pages: pages.map(page => {
-                  return {
-                    condition: {
-                      allMatches: page.condition.allMatches,
-                      id: page.condition.id || undefined,
-                      statements: formatStatements(page.condition.statements),
-                    },
-                    pageId: page.pageId || undefined,
-                    template: page.template,
-                  }
-                }),
+                metaTags:
+                  metaTagDescription || metaTagKeywords
+                    ? {
+                        description: metaTagDescription,
+                        keywords: (metaTagKeywords || []).map(
+                          ({ value }) => value
+                        ),
+                      }
+                    : undefined,
+                pages:
+                  pages &&
+                  pages.map(page => {
+                    return {
+                      condition: {
+                        allMatches: page.condition.allMatches,
+                        id: page.condition.id || undefined,
+                        statements: formatStatements(page.condition.statements),
+                      },
+                      pageId: page.pageId || undefined,
+                      template: page.template,
+                    }
+                  }),
                 path,
-                routeId: routeId || generateNewRouteId(interfaceId, path),
+                routeId: routeId || generateNewRouteId(interfaceId!, path!),
```

Comment:
:pick: It may be better to add checks for those variables instead of using `!`, since they've been typed as possibly `null` and the function doesn't expect `null` arguments; redundancy beats runtime errors, IMHO.